### ISSUE TITLE
Review design of Ockam.Services.API.Endpoint.handle_request/2 and ancestors

### ISSUE THREAD

**Issue by shanesveller** [build-trust/ockam#4706]
### Current behavior

The current design of [Ockam.Services.API.Endpoint](https://github.com/build-trust/ockam/blob/43d2ecf6c5d6da8d8a86ebfcf9576d619f9d8650/implementations/elixir/ockam/ockam_services/lib/services/api/endpoint.ex) has some prior design assumptions that have been questioned during recent code review. Further discussion of new designs was deferred.

https://github.com/build-trust/ockam/pull/4660#discussion_r1163309703

### Desired behavior

Review the implicated code together and consider whether any improvements or desirable refactors are possible, then file issues with designs and prioritize.

**Comment by hairyhum** [comment]
When designing an "API request handler" approach, it would make sense to separate the runtime of the handler execution from the handler function implementation. This allows to define a handler as a function from Request to Response and manage the actual execution in the endpoint level. 
However this requires some additional configuration format for the endpoint to instruct it to run some handlers in parallel, which can be implemented differently, for example we might want to run them all synchronized, or run some of them synchronized, or to run all in parallel, or to have some pool of runners to run in parallel.
Relatively simple and a flexible approach would be to have a pool of runners,  if the pool size is 1 - all requests are synchronized if the pool size is -> ∞ then all handlers run in parallel.
The pool size may be configured per handler or per endpoint, which is TBD.
